src/passes/RemoveUnusedNames.cpp

```diff
@@ -37,16 +37,11 @@ struct RemoveUnusedNames
   std::map<Name, std::set<Expression*>> branchesSeen;
 
   void visitExpression(Expression* curr) {
-    if (auto* block = curr->dynCast<Block>()) {
-      visitBlock(block);
-      return;
-    }
-    if (auto* loop = curr->dynCast<Loop>()) {
-      visitLoop(loop);
-      return;
-    }
```

Comment:
If this function does not call `visitBlock` and `visitLoop`, and this is a `UnifiedExpressionVisitor`, who calls them? Are they ever called?

Reply:
The mechanism of `UnifiedExpressionVisitor` is to define `visitFoo(Foo* curr) { visitExpression(curr); }` for each class. So they all redirect to the main `visitExpression` method. But it is possible to implement some and override that.